src/dali/abstract_pair_converter_plugin.py

```diff
@@ -12,25 +12,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from datetime import datetime
-from typing import Dict, NamedTuple, Optional, cast
+import traceback
+from datetime import datetime, timedelta
+from json import JSONDecodeError
+from typing import Any, Dict, List, NamedTuple, Optional, cast
 
-from rp2.rp2_decimal import RP2Decimal
+import requests
+from requests.models import Response
+from requests.sessions import Session
+from rp2.rp2_decimal import ZERO, RP2Decimal
 from rp2.rp2_error import RP2TypeError
 
 from dali.cache import load_from_cache, save_to_cache
 from dali.configuration import HISTORICAL_PRICE_KEYWORD_SET
 from dali.historical_bar import HistoricalBar
 from dali.logger import LOGGER
 
+# exchangerates.host keywords
+_SUCCESS = "success"
+_SYMBOLS = "symbols"
+_RATES = "rates"
+
+_DAYS_IN_S = 86400
+
 
 class AssetPairAndTimestamp(NamedTuple):
     timestamp: datetime
     from_asset: str
     to_asset: str
+    exchange: str
 
 
 class AbstractPairConverterPlugin:
```

Comment:
Ok, I guess this is used for caching for converter plugins that can access multiple exchanges. Maybe it doesn't need to be in a base class at all? I can just put it in CCXT, since I don't think we will have any other plugins with that functionality.

Reply:
> Ok, I guess this is used for caching for converter plugins that can access multiple exchanges. Maybe it doesn't need to be in a base class at all? I can just put it in CCXT, since I don't think we will have any other plugins with that functionality.

So I guess there are two options:
* put the code in the CCXT pair converter (OK with me: we could extract it into an abstract subclass later, if needed)
* add a new abstract subclass of AbstractPairConverter() containing this logic and have the CCXT pair converter derive from it

I think we can go for the first option for the time being.